packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -1796,114 +1762,68 @@ String camelCaseToHyphenatedName(String word) {
-
-/// A mixin that helps dump string and [DiagnosticsNode] representations of
-/// trees.
-///
-/// Use this class to implement [TreeDiagnostics] any time it is possible to use
-/// mixins.
-abstract class TreeDiagnosticsMixin implements TreeDiagnostics {
-  // This class is intended to be used as a mixin, and should not be
-  // extended directly.
-  factory TreeDiagnosticsMixin._() => null;
+  const Diagnosticable();
 
   /// A brief description of this object, usually just the [runtimeType] and the
   /// [hashCode].
   ///
   /// See also:
   ///
-  ///  * [toStringShallow], for a detailed description of the object.
-  ///  * [toStringDeep], for a description of the subtree rooted at this object.
-  @override
-  String toString() => describeIdentity(this);
+  ///  * [toString], for a detailed description of the object.
+  String toShortDescription() => describeIdentity(this);
 
-  /// Returns a one-line detailed description of the object.
-  ///
-  /// This description is often somewhat long. This includes the same
-  /// information given by [toStringDeep], but does not recurse to any children.
-  ///
-  /// The [toStringShallow] method can take an argument, which is the string to
-  /// place between each part obtained from [debugFillProperties]. Passing a
-  /// string such as `'\n '` will result in a multiline string that indents the
-  /// properties of the object below its name (as per [toString]).
-  ///
-  /// See also:
-  ///
-  ///  * [toString], for a brief description of the object.
-  ///  * [toStringDeep], for a description of the subtree rooted at this object.
-  String toStringShallow([String joiner = '; ']) {
-    final StringBuffer result = new StringBuffer();
-    result.write(toString());
-    result.write(joiner);
-    final List<DiagnosticsNode> properties = <DiagnosticsNode>[];
-    debugFillProperties(properties);
-    result.write(
-      properties.where((DiagnosticsNode n) => !n.hidden).join(joiner),
-    );
-    return result.toString();
+  @override String toString() {
+    return toDiagnosticsNode(style: DiagnosticsTreeStyle.singleLine).toString();
   }
 
-  /// Returns a string representation of this node and its descendants.
-  ///
-  /// The [toStringDeep] method takes arguments, but those are intended for
-  /// internal use when recursing to the descendants, and so can be ignored.
-  ///
-  /// See also:
-  ///
-  ///  * [toString], for a brief description of the object but not its children.
-  ///  * [toStringShallow], for a detailed description of the object but not its
-  ///    children.
-  @override
-  String toStringDeep([String prefixLineOne = '', String prefixOtherLines = '']) {
-    return toDiagnosticsNode().toStringDeep(prefixLineOne, prefixOtherLines);
+  /// Default diagnostics tree style to use.
+  @protected
+  DiagnosticsTreeStyle get debugDefaultDiagnosticsTreeStyle {
```

Comment:
since this class' members will appear on a whole bunch of APIs, I would rather we kept the number of members (even protected ones) to a strict minimum. Ideally just toString, toStringShort (named that so that it sorts after and next to toString), toDiagnosticsNode (which is inevitable since it's the point of the class), and debugFillProperties (which is the one thing we want people to override).